### ISSUE TITLE
Rename master branch to main

### ISSUE THREAD

**Issue by mp911de** [lettuce-io/lettuce-core#1308]
Until recently, I haven’t fully considered the racist connotations of terms like whitelist and blacklist and master. While we had efforts a while ago (#845), we want to reduce wording that is connotated with oppressive (racist) language hence we should revisit branch names and documentation for wording improvements.

**Comment by perlun** [comment]
Very late to the party, but I happened to see the `UPSTREAM`-terminology being used now while looking at the Lettuce Wiki for Lettuce 6. I'm not so sure this deprecation fully makes sense @mp911de. Lettuce is a Redis driver and AFAIK, Redis uses the "master-replication" terminology, as in https://redis.io/topics/replication. Note how that page quite elegantly avoids using the word "slave", which I think is the more potentially offensive language.

"Upstream", AFAIK, is not a word that is being actively used in Redis terminology (please correct me if I'm wrong!), so us using it feels like it does more harm than good.

(More upstream Redis discussion can be found in this issue: https://github.com/redis/redis/issues/5335, as well as @antirez blog post from a couple of years ago: http://antirez.com/news/122)

----

**TL;DR**: As far as I understand it, Redis uses the "master-replication" terminology. I think we are doing ourselves and the community harm if we deviate from this terminology, it just confusing. If you think this terminology is wrong/bad, I think we should try to get Redis to stop using it as well.

**Comment by mp911de** [comment]
Thanks for chiming in @perlun. We were a bit more progressive on the client side and not fully sure how using `Upstream` instead of `Master` will turn out. We always have the chance to reinstate the old master-replica wording. Can you file a ticket and add your comment as description?

**Comment by perlun** [comment]
> We were a bit more progressive on the client side and not fully sure how using `Upstream` instead of `Master` will turn out. We always have the chance to reinstate the old master-replica wording. Can you file a ticket and add your comment as description?

Thanks for listening, appreciated. :+1: 

I filed https://github.com/lettuce-io/lettuce-core/issues/1518 about this now, let's move the discussion there.
